packages/venia-ui/lib/components/Image/image.js

```diff
@@ -65,8 +65,7 @@ const Image = props => {
 
     const classes = mergeClasses(defaultClasses, propsClasses);
     const containerClass = `${classes.root} ${classes.container}`;
-    const isLoadedClass = isLoaded ? classes.loaded : classes.notLoaded;
-    const imageClass = `${classes.image} ${isLoadedClass}`;
+    const imageClass = `${classes.image} ${classes.image_positioned}`;
```

Comment:
So I know this snuck through the last couple PRs, but this is really not how we're supposed to be using classnames. Applying multiple classnames in this way creates ambiguity, since the source order of CSS *rulesets* matters but the order of *classnames* does not (and the source order of rulesets is beyond our control). That ambiguity manifests as bugs.

Instead, each element we create should have **only one** classname applied to it (e.g., `classes.image`), and we should use `composes` to ensure that all the right styles have been included. Of course, CSS Modules may map its keys to multiple classnames (as output), but [proper use of `composes`](https://github.com/css-modules/css-modules#composing-from-other-files) ensures that they're deterministic.

Reply:
I actually did change this but ran into:

Callers can (and do) pass in overrides to `classes.image` and `classes.root`.

I tried setting `classes.container` to compose `root` (and `classes.image_positioned` to compose `image`), but they weren't picking up the passed-in overrides.

Let me know if that doesn't make sense - I'd love to fix this but I'm not sure how.

Reply:
You're correct. When overriding classnames, a caller must pass in overrides for the final classnames being applied, not the "base" classnames. Take the following sample component:

```jsx
const Foo = props => {
  const [isOpen, setIsOpen] = useState(false)
  const classes = mergeClasses(defaultClasses, props.classes)
  const rootClass = isOpen ? classes.root_visible : classes.root_hidden

  return <div className={rootClass}>{props.children}</div>
}
```

```css
.root {
  color: red;
  display: grid;
}

.root_visible {
  composes: root;
}

.root_hidden {
  composes: root;
  display: none;
}
```

In this example, the classnames that make up `Foo`'s public API are `classes.root_hidden` and `classes.root_visible`, the only keys that `Foo` applies. The component does not apply `classes.root` to anything, so that classname is not part of `Foo`'s API at all. It's essentially a mixin that the CSS module may opt to use to avoid some repetition—nothing more.

Reply:
Hoping we can split this out into a separate ticket so we can resolve this P1 with minimal impact.